### PR TITLE
add session window section to `sql-function-time-window.md`

### PR DESCRIPTION
**Pull request by stdrc**
## Info

- **Description**

  - Add some descriptions and examples for Session Window in Time Window Function page.
  - Re-phrase some paragraphs in Time Window Function page.

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves [ Provide a link to the relevant doc issue here, if applicable. ]

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

**Review by neverchanje — APPROVED** [review]
Looks great! Thanks!

File: docs/sql/functions-operators/sql-function-time-window.md [hunk 1]
```diff
@@ -9,150 +9,200 @@ title: Time window functions
...
+The syntax of the `tumble()` is as follows:
 
 ```sql
 SELECT [ ALL | DISTINCT ] [ * | expression [ AS output_name ] [, expression [ AS output_name ]...] ]
-FROM TUMBLE ( table_or_source, start_time, window_size [, offset ] );
+FROM TUMBLE ( table_or_source, time_col, window_size [, offset ] );
 ```
 
-- *start_time* can be in either timestamp or timestamp with time zone format.
+- *table_or_source* specifies the table/source/materialized view that need to be assigned with time windows.
```
Comment (WanYixian) [inline docs/sql/functions-operators/sql-function-time-window.md]:
```suggestion
- *table_or_source* specifies the table/source/materialized view that needs to be assigned with time windows.
```

File: docs/sql/functions-operators/sql-function-time-window.md [hunk 2]
```diff
@@ -9,150 +9,200 @@ title: Time window functions
...
+ 4       | 1004    | 2022-07-01 22:03:00 | 2022-07-01 22:02:00 | 2022-07-01 22:04:00
+ 5       | 1005    | 2022-07-01 22:05:00 | 2022-07-01 22:04:00 | 2022-07-01 22:06:00
+ 6       | 1006    | 2022-07-01 22:06:00 | 2022-07-01 22:06:00 | 2022-07-01 22:08:00
 ```
 
-
 ## `hop()` time window function
 
-Hopping windows are scheduled time intervals. A hopping window consists of three time-related parameters: start time, hop size, and window size.
+The `hop()` time window function also assigns each row a time window with fixed size, which is very similar to `tumble()`, except that the assigned time windows may overlap.
```
Comment (WanYixian) [inline docs/sql/functions-operators/sql-function-time-window.md]:
```suggestion
The `hop()` time window function also assigns each row a time window with a fixed size, which is very similar to `tumble()`, except that the assigned time windows may overlap.
```

File: docs/sql/functions-operators/sql-function-time-window.md [hunk 3]
```diff
@@ -9,150 +9,200 @@ title: Time window functions
...
+See below for the syntax of the `hop()` time window function.
 
 ```sql
 SELECT [ ALL | DISTINCT] [ * | expression [ AS output_name ] [, expression [ AS output_name ]...] ]
-FROM HOP ( table_or_source, start_time, hop_size, window_size [, offset ]);
+FROM HOP ( table_or_source, time_col, hop_size, window_size [, offset ]);
 ```
 
-- *start_time* can be in either timestamp or timestamp with time zone format.
+- *table_or_source* specifies the table/source/materialized view that need to be assigned with time windows.
```
Comment (WanYixian) [inline docs/sql/functions-operators/sql-function-time-window.md]:
```suggestion
- *table_or_source* specifies the table/source/materialized view that needs to be assigned with time windows.
```

File: docs/sql/functions-operators/sql-function-time-window.md [hunk 4]
```diff
@@ -186,6 +236,37 @@ The result looks like this:
  2022-07-01 22:06:00 | 2022-07-01 22:08:00 | 1           | 8
 ```
 
+### Session window aggregations
+
+Below is an example of aggregation over session windows. In this example, we want get the number of unique products viewed by each user in session gapped by 5 minutes interval, based on the example data in previous [Session windows](#session-windows) section.
```
Comment (WanYixian) [inline docs/sql/functions-operators/sql-function-time-window.md]:
```suggestion
Below is an example of aggregation over session windows. In this example, we want to get the number of unique products viewed by each user in session gapped by 5 minutes interval, based on the example data in previous [Session windows](#session-windows) section.
```

**Review by WanYixian — APPROVED** [review]
Thank you RC! BTW, saw both `tumbling window` and `tumble window` in the doc, shall we make the terminology consistent, and which one is more commonly accepted?

**Comment by stdrc** [comment]
> Thank you RC! BTW, saw both `tumbling window` and `tumble window` in the doc, shall we make the terminology consistent, and which one is more commonly accepted?

Both are fine, and I saw documents of some other platforms mixing these two terminologies. So I guess it's not a problem. I changed the list item at the beginning only for visually harmony with "session window".

**Pull request merged**